### PR TITLE
Make Jenkins implement AbstractAsyncContextManager + Improve tests/README to use it

### PR DESCRIPTION
**Pull request by PaarthShah**
I was using this for a project, and I didn't like having to write `try...finally: await jenkins.close()`.
This is the idiomatic way to setup/close resources such as this, and I'd love to see this merged/published to pypi if possible! Tests all pass locally (including with ~~3.7~~ 3.8, the minimum supported version of python for this project).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -44,17 +44,11 @@ Start new build:
...
-    loop = asyncio.get_event_loop()
-    try:
-        loop.run_until_complete(example())
-    finally:
-        loop.run_until_complete(jenkins.close())
-        loop.close()
+        async with aiojenkins.Jenkins('http://your_server/jenkins', 'user', 'password') as jenkins:
+            await jenkins.builds.start('job_name', dict(parameter='test'))
+
+    asyncio.run(example())
```
Comment (PaarthShah) [inline README.rst:68]: Available as of python3.7, which this project ~~lists~~ formerly listed as its minimum version
https://docs.python.org/3/library/asyncio-runner.html#running-an-asyncio-program

File: aiojenkins/jenkins.py [hunk 2]
```diff
@@ -64,7 +65,7 @@ async def close(self) -> None:
         await self.session.close()
 
 
-class Jenkins:
+class Jenkins(AbstractAsyncContextManager):
```
Comment (PaarthShah) [inline aiojenkins/jenkins.py:70]: Subclassing this automatically provides the default `__aenter__` method (which returns `self`)

File: setup.cfg [hunk 3]
```diff
@@ -7,7 +7,7 @@ passenv = *
 deps = -r{toxinidir}/requirements.txt
        -r{toxinidir}/tests/requirements.txt
 commands = python3 -B -m pytest {tty:--color=yes} {posargs} -v --mypy \
-           --cov=aiojenkins --cov=tests --cov-report=term --ignore docs
+           --cov=aiojenkins --asyncio-mode=auto --cov=tests --cov-report=term --ignore docs
```
Comment (PaarthShah) [inline setup.cfg:9]: Removes the need to include `@pytest.mark.asyncio` everywhere

File: tests/test_jenkins.py [hunk 4]
```diff
@@ -184,5 +171,3 @@ async def test_make_jenkins_version(jenkins, aiohttp_mock):
     )
     version = await jenkins.get_version()
     assert version == JenkinsVersion(major=2, minor=346, patch=1, build=4)
-
-    await jenkins.close()
```
Comment (PaarthShah) [inline tests/test_jenkins.py:188]: This seemed like a stray call to `close` and tests passed without it

**Comment by PaarthShah** [comment]
@pbelskiy Would this be good to merge along with a version bump? 👀

File: tests/test_jenkins.py [hunk 5]
```diff
@@ -118,17 +111,12 @@ async def request(*args, **kwargs):  # pylint: disable=unused-argument
 
     retry = {'total': 5, 'statuses': [HTTPStatus.INTERNAL_SERVER_ERROR]}
 
-    try:
-        jenkins = Jenkins(get_host(), get_user(), get_password(), retry=retry)
-
+    async with Jenkins(get_host(), get_user(), get_password(), retry=retry) as jenkins:
```
Comment (pbelskiy) [inline tests/test_jenkins.py:115]: Why did you change it?

File: setup.cfg (hunk 3 above)
Comment (pbelskiy) [reply setup.cfg:9]: Looks good

File: README.rst (hunk 1 above)
Comment (pbelskiy) [reply README.rst:68]: Okay, but please add example with context manager as additional to just simple definition

File: tests/test_jenkins.py (hunk 5 above)
Comment (PaarthShah) [reply tests/test_jenkins.py:115]: Functionally-speaking, it's identical (since either way the `.close()` is being called after the `with` block vs the `try` block), but this this usage then follows the best practice of using an async context manager to auto-close resources.

If it's significant to do so, I can revert it with no issues

File: README.rst (hunk 1 above)
Comment (PaarthShah) [reply README.rst:68]: Sounds good, will push that shortly

Comment (PaarthShah) [reply README.rst:68]: Done

File: tests/test_jenkins.py (hunk 5 above)
Comment (pbelskiy) [reply tests/test_jenkins.py:115]: We need both tests, as it was and with context manager

Comment (PaarthShah) [reply tests/test_jenkins.py:115]: Done, re-added them back

**Review by pbelskiy — APPROVED** [review]
